src/hooks/useDimensions.ts

```diff
@@ -0,0 +1,29 @@
+import { MutableRefObject, useLayoutEffect, useMemo, useState } from "react";
+
+interface DimensionState {
+  width: number;
+  height: number;
+}
+
+export const useDimensions = (ref: MutableRefObject<HTMLElement>) => {
```

Comment:
So we already have an AutoSizer component how do you feel about this hook approach should we be using it instead? If so I think we should consolidate on one approach for calculating the size of elements/ the page.

https://github.com/evergreen-ci/spruce/blob/a02985f93b266377640380220b1d3142cb5e0ae8/src/components/HistoryTable/HistoryTable.tsx#L84

Reply:
Makes sense to me thanks for the explanation!